### PR TITLE
Add track events for quick link ribbon

### LINKED ISSUE #16272: Add track events for Quick Link ribbon
Add tracks events "quick_action_ribbon_tapped" with properties: "button" = "stats|pages|posts|media" and "tab_source" = "dashboard|site_menu" . See `MySiteViewModel.trackWithTabSourceIfNeeded`

### PR DESCRIPTION
**Pull request by AjeshRPai**
Fixes #16272 

This PR adds the support for tracking events when tapping the quick link ribbon buttons. Pages/Posts/Media/Stats

Note: Only 1 approval is needed. Thanks

**To test:**
- Launch the app
- Navigate to Me -> App Settings -> Debug Settings
- Enable MySiteDashboardTabsFeatureConfig and restart the app
- Navigate to My Site View
- Tap on the following (returning back to My Site View after each click):  
   - All Quick Link buttons (Stats, Pages, Posts, and Media) 
-  Swipe down on the My Site View to request a pull-to-refresh
- View the logs and note that "tab_source=dashboard" is included in the properties of the track event.  
```
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"pages"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"posts"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"media"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"stats"}
```

## Regression Notes
1. Potential unintended areas of impact
Events for quick link ribbon buttons are not tracked. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests  

3. What automated tests I added (or what prevented me from doing so)
Added tests in `MySiteViewModelTest.kt`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt [hunk 1]
```diff
@@ -2163,6 +2163,82 @@ class MySiteViewModelTest : BaseUnitTest() {
...
+
+    @Test
+    fun `given tabs are disabled, when quick link ribbon media tapped, then track with tab source is not requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = false, isMySiteTabsBuildConfigEnabled = false)
+
+        requireNotNull(quickLinkRibbonMediaClickAction).invoke()
+
+        assertThat(trackWithTabSource).isEmpty()
+        assertThat(analyticsTrackerWrapper.track(Stat.QUICK_LINK_RIBBON_MEDIA_TAPPED))
+    }
```
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: ❓ I'm wondering if we need `given tabs are disabled` test cases. Original quick actions are shown even when tabs are disabled. But the `quick link ribbon` is available only on `Home`/ `Dashboard` tab. We will never be able to execute the`when clause` in the app as quick link ribbon is not built when tabs are disabled. 

We might want to add test cases in the `org.wordpress.android.ui.mysite.cards.CardsBuilder` whether a quick link ribbon  is built. 

Wdyt?

**Review by ashiagr — COMMENTED** [review]
Thanks for adding these trackings, @AjeshRPai! I can confirm that events are tracked as described:

```
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"pages"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"posts"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"media"}
🔵 Tracked: quick_action_ribbon_tapped, Properties: {"default_tab_experiment":"nonexistent","tab_source":"dashboard","button":"stats"}
```
But I left one question on the test cases. Lmkwyt.

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: > I'm wondering if we need given tabs are disabled test cases. Original quick actions are shown even when tabs are disabled. But the quick link ribbon is available only on Home/ Dashboard tab. We will never be able to execute thewhen clause in the app as quick link ribbon is not built when tabs are disabled.


Yup, you are right. 💯 . What was I thinking 🤔 . I have removed the disabled test cases in the commit. [65907ef](https://github.com/wordpress-mobile/WordPress-Android/pull/16284/commits/65907efbcd61a0d12edfb97b1d1d2878cb6a07eb)


>We might want to add test cases in the org.wordpress.android.ui.mysite.cards.CardsBuilder whether a quick link ribbon is built.

I have added the test case for the condition `if tabs are disabled, then quick link should not be built` in `CardsBuilderTest.kt`. Please find the details below. 

Also I am not sure how those tabs disabled tests passed in `MyViewModelTest.kt`. I believe the initialisation for those tests was wrong, I didn't go deeper on how those succeeded as those tests were not needed anyways. 

Screenshot 2022-04-08 at 6 33 35 PM

Comment (ashiagr) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: I can see card builder tests for the ribbon, so we're good. Thank you for removing redundant test cases. 😄

**Review by ashiagr — APPROVED** [review]
LGTM

:shipit:

**Pull request merged**